tests/test_product.py

```diff
@@ -532,6 +543,123 @@ def test_dot_out_error(self, shape_pair):
                 numpy.dot(a, b, out=np_out)
 
 
+class TestInner:
+    def setup_method(self):
+        numpy.random.seed(42)
+
+    @pytest.mark.parametrize("dtype", get_all_dtypes())
+    def test_inner_scalar(self, dtype):
+        a = 2
+        b = numpy.array(numpy.random.uniform(-5, 5, 10), dtype=dtype)
+        ib = dpnp.array(b)
+
+        result = dpnp.inner(a, ib)
+        expected = numpy.inner(a, b)
+        if dtype in [numpy.int32, numpy.float32, numpy.complex64]:
+            assert_dtype_allclose(result, expected, check_only_type_kind=True)
+        else:
+            assert_dtype_allclose(result, expected)
+
+        result = dpnp.inner(ib, a)
+        expected = numpy.inner(b, a)
+        if dtype in [numpy.int32, numpy.float32, numpy.complex64]:
```

Comment:
@antonwolfy just wanted to highlight this difference between `NumPy` and `dpnp` when one of the input is scalar to make sure we are okay with it.

Reply:
It's quite the same like with `sum` function done, there is also different type promotion rules between numpy and dpnp.
I believe currently we can leave that like it is. Later, probably it will be changed, especially when NEP-50 will be implemented by numpy.